Feat: 엔티티 클래스 생성 (#12)

Ju-yeop opened the pull request:
## 🧑🏻‍💻 PR 작업 내역 요약
배틀관련 엔티티를 생성했습니다.

## 📋 변경 사항
- **V1** - Entity
    - Duel
    - DuelEntry
    - FcmToken
    - Notification
    - Team
    - TeamBattle
    - TeamBattleEntry
    - TeamEntry
   
- **V2** - Entity
    - MatchEntry
    - Match
    - UserMatch
    - FcmToken
    - Notification

 - User - 연관관계 매핑을 위한 엔티티 (컬럼 생성X)
 
## 🔍 Code Review
코드 리뷰 시에 주요 확인 사항을 작성해주세요.

## 연결된 이슈
#12

Ju-yeop committed:
Feat: User Entity 추가 (#12)
연관관계 매핑을 위한 엔티티 클래스 생성
추후 수정 필요

Ju-yeop committed:
Feat: Goal Enum 추가 (#12)

Ju-yeop committed:
Feat: Period Enum 추가 (#12)

Ju-yeop committed:
Feat: Strength Enum 추가 (#12)

Ju-yeop committed:
Feat: BattleType Enum 추가 (#12)

Ju-yeop committed:
Feat: BattleStatus Enum 추가 (#12)

Ju-yeop committed:
Feat: TeamBattle Entity 추가 (#12)

Ju-yeop committed:
Feat: Team Entity 추가 (#12)

Ju-yeop committed:
Feat: TeamBattleEntry Entity 추가 (#12)

Ju-yeop committed:
Feat: TeamEntry Entity 추가 (#12)

Ju-yeop committed:
Feat: Duel Entity 추가 (#12)

Ju-yeop committed:
Feat: DuelEntry Entity 추가 (#12)

Ju-yeop committed:
Feat: Notification Entity 추가 (#12)

Ju-yeop committed:
Feat: FcmToken Entity 추가 (#12)

Ju-yeop committed:
Refact: JoinColumn 이름 변경 (#12)

kaley0421 commented on src/main/java/com/dnd/Exercise/domain/notification/entity/Notification.java:
```diff
@@ -0,0 +1,47 @@
+    private Boolean isRead;
+
+    @Enumerated(EnumType.STRING)
+    private BattleType battleType;
```
BattleType 에 따라서 아래의 team, user, duel 필드 중 하나에만 값이 들어가는 식으로 동작하는 건가요??
테이블을 별도로 나누는 것보다 그냥 이렇게 한번에 관리하는게 더 편할까요?!?

Ju-yeop replied:
네 맞아요!!
저도 조인전략으로 해서 테이블을 분리할까 고민해봤는데
각 유형들이 고유한 필드들을 많이 가지지 않고 알림 유형별 동작이 모두 동일해서 단일 테이블로 구현했습니다!
쿼리가 단순해지는 장점도 있을 것 같아요!

kaley0421 commented on src/main/java/com/dnd/Exercise/domain/team/entity/Team.java:
```diff
@@ -0,0 +1,56 @@
+
+    private Boolean isActive;
+
+    private Period period;
```
Period 에 @Enumerated 빠진것 같아요!

Ju-yeop replied:
앗 감사합니다! 추가해 놓겠습니다

kaley0421 commented on src/main/java/com/dnd/Exercise/domain/team/entity/Team.java:
```diff
@@ -0,0 +1,56 @@
+
+    private String rule;
+
+    private int teamSize;
```
팀원 모집중/모집 완료 여부는 따로 컬럼을 두지 않고 teamSize 와 현재까지 모인 팀원 수를 비교해서 구분하면 될까요??

Ju-yeop replied:
코드 작성할 때는 그럴 생각이었는데 쿼리 할 때마다 조인이 필요해져서 성능 상 안좋겠네요
모집중/모집 완료 여부를 넣는 것도 좋지만 <현재 팀원 수> 컬럼을 추가하는건 어떨까요
UI상으로도 현재 팀원 수 보여줘야 하는 부분이 있고 모집중/모집완료 여부는 <현재 팀원 수>와 <teamSize>를 비교해서  보여줄 수 있을 것 같아요

kaley0421 replied:
현재 팀원 수 좋은것 같습니다!!

Ju-yeop committed:
Fix: DB 설계 변경으로 인한 엔티티 삭제 (#12)

Ju-yeop committed:
Feat: DB 설계 변경으로 인한 엔티티 추가 및 수정 (#12)
- Team 워딩 변경 -> Match(팀, 팀배틀, 1대1 배틀)
- ERD 참고 (https://www.erdcloud.com/d/23SfpuyhYQcEFrFsR)
